pysam/libcalignmentfile.pyx

```diff
@@ -273,7 +307,193 @@ cdef bam_hdr_t * build_header_from_list(reference_names,
             raise MemoryError("could not allocate {} bytes".format(strlen(ctext), sizeof(char)))
         memcpy(dest.text, ctext, strlen(ctext))
 
-    return dest
+    return new_header
+
+
+cdef class AlignmentHeader(object):
+    """header information for a :class:`AlignmentFile` object"""
+
+    # See makeVariantHeader for C constructor
+    def __cinit__(self):
+        self.ptr = NULL
+
+    # Python constructor
+    def __init__(self):
+        self.ptr = bam_hdr_init()
+        if not self.ptr:
+            raise ValueError('cannot create AlignmentHeader')
+
+    def __dealloc__(self):
+        if self.ptr:
+            bam_hdr_destroy(self.ptr)
+            self.ptr = NULL
+
+    def __bool__(self):
+        return self.ptr != NULL
+
+    def copy(self):
+        return makeAlignmentHeader(bam_hdr_dup(self.ptr))
+
+    def clear(self):
```

Comment:
Strongly recommend not allowing a header to be cleared unless deallocated.  Allowing `ptr = NULL` in a live header will do more than invalidate objects, it will cause segfaults when accessing those objects.